lib/common.functions.php

```diff
@@ -864,3 +864,63 @@ function url2absolute($baseurl, $relativeurl) {
 	}
 	return implode('/', $returnpath);
 }
+
+/**
+ * Shorthand for a curl request.
+ * @param $url String The url for the request
+ * @param array $options curl options
+ * @return mixed The curl_exec result
+ */
+function curl_request($url, $options = []) {
+	$curl = curl_init($url);
+	curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+	curl_setopt_array($curl, $options);
+	return curl_exec($curl);
+}
+
+/**
+ * Follow an url to its final location taking http redirects and meta refreshes into account.
+ *
+ * @param String $url The url to follow to its final location
+ * @param array $options A curl_setopt_array compatible array
+ * @return String The final url location
+ */
+function curl_follow_location($url, $options = []) {
+	$curl = curl_init($url);
+	curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+	curl_setopt($curl, CURLOPT_FOLLOWLOCATION, true);
+	curl_setopt_array($curl, $options);
+	$xpath = init_xpath(curl_exec($curl));
+	$location = curl_getinfo($curl, CURLINFO_EFFECTIVE_URL);
+
+	$refresh = $xpath->query('//meta[translate(@http-equiv, "abcdefghijklmnopqrstuvwxyz", "ABCDEFGHIJKLMNOPQRSTUVWXYZ")="REFRESH"]');
+	if ($refresh->length > 0) {
+		preg_match('/(?<=url=)(.*)/i', $refresh->item(0)->getAttribute('content'), $matches);
+		$refreshUrl = trim($matches[0]);
+
+		// clk.tradedoubler.com specific meta-refresh workaround
+		if (endsWith($location, 'f=0')) {
+			return curl_follow_location(str_replace('f=0', '&f=0', $location));
+		}
+
+		if (!startsWith($refreshUrl, 'http')) {
+			$refreshUrl = http_build_url($location, $refreshUrl, HTTP_URL_REPLACE | HTTP_URL_JOIN_PATH);
```

Comment:
Wat is het verschil tussen `http_build_url` en `url2absolute`?

Reply:
Goede vraag, in de praktijk zou url2absolute ook kunnen werken hier denk ik.
de http_build_url die hier gebruikt wordt is een drop-in replacement voor de http_build_url uit de pecl_http php library: http://php.net/manual/fa/function.http-build-url.php
Deze functie kan dus ook omgaan met arrays die door parse_url gegenereerd zijn en is flexibeler wat betreft merge opties.